pkg/istio/rbac/v2/provider_test.go

```diff
@@ -42,37 +42,159 @@ var (
-
-	convertedCR := p.ConvertAthenzModelIntoIstioRbac(domainRBAC, onboardedService.Name, labels["app"])
-	expectedCR := getExpectedCR()
-	// when there are multiple assertions matched with svc,
-	// base on which assertion will be processed from the athenzModel mapping, order of the authz rule generated
-	// may be different each time. Using a workaround here to sort the configSpec in alphabetical order of first
-	// method in each rule.
-	configSpec := (convertedCR[0].Spec).(*v1beta1.AuthorizationPolicy)
-	sort.Slice(configSpec.Rules, func(i, j int) bool {
-		return configSpec.Rules[i].To[0].Operation.Methods[0] < configSpec.Rules[j].To[0].Operation.Methods[0]
-	})
-	convertedCR[0].Spec = configSpec
-
-	assert.EqualValues(t, expectedCR, convertedCR, "converted authz policy should be equal")
+	tests := []struct {
+		name                string
+		inputAthenzDomain   zms.SignedDomain
+		inputService        *k8sv1.Service
+		expectedAuthzPolicy []model.Config
+	}{
+		{
+			name:                "should create empty authz policy spec for service which doesn't have roles and policies defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(false, false, false, false),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+		{
+			name:                "should create empty authz policy spec for service with empty role and no policies defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(true, false, false, false),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+		{
+			name:                "should create empty authz policy spec for service with one member in role and no policies defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(true, true, false, false),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+		{
+			name:                "should create empty authz policy spec for service with no role and empty policy defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(false, false, true, false),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+		{
+			name:                "should create empty authz policy spec for service with empty role and empty policy defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(true, false, true, false),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+		{
+			name:                "should create authz policy spec with role for service with no role and policy defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(false, false, true, true),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicyWithRole(),
```

Comment:
minor: could we call this `getExpectedAuthzPolicyWithRole` since it is no longer an empty authz policy that is being returned

Reply:
Sure, updated.